ray-operator/test/sampleyaml/support.go

```diff
@@ -74,3 +77,55 @@ func AllPodsRunningAndReady(pods []corev1.Pod) bool {
 	}
 	return true
 }
+
+func IsRayReady(t Test, pods []corev1.Pod) bool {
```

Comment:
I think this function should be renamed to something like `SubmitJobsToAllPods`, and the function signature should return another function, taking `gomega.Gomega` as input. And we make assertions in this function.

This is the [Category 3: Making assertions in the function passed into Eventually](https://onsi.github.io/gomega/#category-3-making-assertions-eminem-the-function-passed-into-codeeventuallycode) in the official gomega doc.

You can reference this file.

https://github.com/ray-project/kuberay/blob/1213d1578ac08a5fc993e2f1eb34fa021fe07749/ray-operator/test/support/ray.go#L58-L64

The caller should use it like

```go
test.Eventually(SubmitJobsToAllPods, TestTimeoutShort).Should(gomega.Succeed())
```

I also found that we misused `Eventually` in multiple places by passing values, instead of functions, to `Eventually`. I'll open another issue to track this.

Reply:
Tracking issue: https://github.com/ray-project/kuberay/issues/2463